### ISSUE TITLE
Portal does not retain calendar information

### ISSUE THREAD

**Issue by acp31** [airbnb/react-dates#149]
When using the calendar with vertical orientation and with a full screen portal after the dates are selected the calendar closes. When the calendar reopens the starting point is at the current date and not the range of dates selected. Is there a way to store the keys that are on the CalendarMonthGrid as well as the prop isVisible so when the calendar is reopened it is set to the selected dates?

**Comment by majapw** [comment]
I had previously noticed this but had not had time to fix this. This is due to the fact that the DayPicker is being mounted and unmounted each time afaict and therefore the state is not being preserved.

**Comment by acp31** [comment]
I was able to make a temporary fix with the vertical orientation with portal. 

`    .DayPicker--vertical {
      position: fixed;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
    }
`
`.DateRangePicker {
  z-index: 3;
}
`
it is not as good as the full screen portal because it does not have the close button in the top right corner.

**Comment by majapw** [comment]
Yeah that's really not as ideal. Probably using the `initialVisibleMonth` prop combined with the `onPrevMonthClick` and `onNextMonthClick` props should be a fix for this particular issue

**Comment by lyf1988** [comment]
Just had same problem, and by idea from @majapw I passed current selected date to `initialVisibleMonth` and it's worked for me.
`initialVisibleMonth={() => this.props.date}`

And I checked the source code, turns out SingleDatePicker will set moment object by default in `initialVisibleMonth`, so I think this may the reason for this problem?
`initialVisibleMonth: () => moment(),`

**Comment by majapw** [comment]
Well, it only does this once, on first open/render, so it makes sense that setting it to your startDate solves your usecase. :)

**Comment by joeduncan** [comment]
Setting a specific month via `initialVisibleMonth` does not have any effect. The corresponding doesn't seem to be working properly either http://airbnb.io/react-dates/?selectedKind=DateRangePicker&selectedStory=with%20month%20specified%20on%20open&full=0&down=1&left=1&panelRight=0&downPanel=kadirahq%2Fstorybook-addon-actions%2Factions-panel

Any ideas?

```
<DateRangePicker
                    id='date-input'
                    startDate={this.state.startDate}
                    endDate={this.state.endDate}
                    initialVisibleMonth={() => moment('2017-10', 'YYYY-MM')}
                    focusedInput={this.state.focused}
                    onDatesChange={this.onDatesChange}
                    onFocusChange={this.onFocusChange}
```
